src/main/java/teammates/ui/controller/StudentProfilePictureAction.java

```diff
@@ -42,8 +43,16 @@ private ActionResult handleRequestWithBlobKey() {
 
     private ActionResult handleRequestWithEmailAndCourse()
             throws EntityDoesNotExistException {
-        String email = getStudentEmailFromRequest();
-        String courseId = getCourseIdFromRequest();
+        String email;
+        String courseId;
+        try {
+            email = getStudentEmailFromRequest();
+            courseId = getCourseIdFromRequest();
+        } catch (InvalidParametersException e) {
+            // potential PMD bug? No stack trace info is lost here
+            throw new EntityDoesNotExistException(e.getMessage()); //NOPMD
```

Comment:
Exceptions are caught further up at the call stack and logged. Maybe what I could do instead is to make the exception message more descriptive. I.e, prepend something like "Message could not be decrypted:" followed by java's original message.